### DISCUSSION TITLE
How To Configure Service for Use With Rtldavis

### DISCUSSION THREAD

**Discussion by gary-hammer** [bellrichm/WeeWX-MQTTSubscribe#172] [Q&A] [answered]
I tried configuring the service to pull a few values from an MQTT feed.
When I did this, rtldavis was terminated.
I'm going to attach the service config and the logs for the issue.

Certain this should work, but unsure why this happened.

[mqtt-rtl-error-log.txt](https://github.com/bellrichm/WeeWX-MQTTSubscribe/files/9929788/mqtt-rtl-error-log.txt)
[service-config.txt](https://github.com/bellrichm/WeeWX-MQTTSubscribe/files/9929789/service-config.txt)

**Comment by bellrichm** [comment]
You are getting a rc of 5 from the connect. This means not authorized. It looks like MQTTSubscribe thinks that the userid is not set, but a password is set. What is set? I don’t need the actual values...
Looks like tis code could be improved.
```
Nov 03 09:36:20 lts weewxd[5656]: weewx[5656] INFO user.MQTTSubscribe: (Service) username is None
Nov 03 09:36:20 lts weewxd[5656]: weewx[5656] INFO user.MQTTSubscribe: (Service) min_delay is 1
Nov 03 09:36:20 lts weewxd[5656]: weewx[5656] INFO user.MQTTSubscribe: (Service) max_delay is 120
Nov 03 09:36:20 lts weewxd[5656]: weewx[5656] INFO user.MQTTSubscribe: (Service) password is set
Nov 03 09:36:20 lts weewxd[5656]: weewx[5656] INFO user.MQTTSubscribe: (Service) Archive topic is None
Nov 03 09:36:20 lts weewxd[5656]: weewx[5656] INFO user.MQTTSubscribe: (Service) binding is loop
Nov 03 09:36:20 lts weewxd[5656]: weewx[5656] INFO user.MQTTSubscribe: (Service) Waiting for MQTT connection.
Nov 03 09:36:20 lts weewxd[5656]: weewx[5656] INFO user.MQTTSubscribe: (Service) Connected with result code 5
```

**Reply by gary-hammer** [reply]
Well, I have password = None
To match the username.

But, no matter, I set a username and password and it runs.
Now, I just need to set the time window wider as all the packets are being rejected

`Nov 03 12:15:27 lts weewxd[6070]: weewx[6070] INFO user.MQTTSubscribe: (Service) TopicManager ignoring record  outside of interval 1667492126.000000 1667492128.000000 1667492125.817981 'barometer': '30.424', 'dateTime':  '1667492125.8179815', 'inHumidity': '40.0', 'inTemp': '74.8', 'usUnits': '1'
`

**Reply by gary-hammer** [reply]
Got it by setting
ignore_start_time = True
ignore_end_time = True

Don't know why use server time did not work though.

**Comment by gary-hammer** [comment]
So, under [[topics]] I set use_server_datetime = True
Still seeing the same ignoring record logs.

**Comment by bellrichm** [comment, marked as answer]
Yeah, unfortunately the time ‘window’ was not the best implementation… Maybe someday it will get redone.

I really appreciate your follow up comments. Hopefully it will help someone in the future.
